File: ide/src/main/java/org/dau/ui/schematic/fx/theme/ThemeApplier.java
package org.dau.ui.schematic.fx.theme;

import com.sun.javafx.css.StyleManager;
import javafx.application.Application;

import static javafx.application.Application.STYLESHEET_CASPIAN;

public interface ThemeApplier {

  static void apply() {
    var classLoader = Thread.currentThread().getContextClassLoader();
    var themeUrl = classLoader.getResource("theme.css");
    Application.setUserAgentStylesheet(STYLESHEET_CASPIAN);
    var styleManager = StyleManager.getInstance();
    if (themeUrl != null) {
      styleManager.addUserAgentStylesheet(themeUrl.toString());
    }
  }
}

File: ide/src/main/java/org/dau/ui/schematic/fx/theme/ThemeApplier.kt
package org.dau.ui.schematic.fx.theme

import com.sun.javafx.css.StyleManager
import javafx.application.Application

import javafx.application.Application.STYLESHEET_MODENA

object ThemeApplier {

  @JvmStatic
  fun apply() {
    val classLoader = Thread.currentThread().contextClassLoader
    val themeUrl = classLoader.getResource("theme.css")
    Application.setUserAgentStylesheet(STYLESHEET_MODENA)
    val styleManager = StyleManager.getInstance()
    if (themeUrl != null) {
      styleManager.addUserAgentStylesheet(themeUrl.toString())
    }
  }
}
